File: Spring/next-board/src/main/java/com/example/nextboard/domain/member/entity/Member.java
package com.example.nextboard.domain.member.entity;

import com.example.nextboard.domain.comment.entity.Comment;
import com.example.nextboard.domain.post.entity.Post;
import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
@AllArgsConstructor
@Builder
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id")
    private Long id;

    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Post> posts = new ArrayList<>();

    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments = new ArrayList<>();

    private String email;
    private String password;
    private String name;

    private int grade;
    private int fellowNumber;

}

File: Spring/kotlin-practice/src/main/kotlin/com/practice/kotlinpractice/domain/member/Member.kt
package com.practice.kotlinpractice.domain.member

import com.practice.kotlinpractice.domain.board.Board
import jakarta.persistence.*
import jakarta.persistence.GenerationType.*
import org.springframework.security.crypto.password.PasswordEncoder

@Entity
class Member(

    val email: String,

    @Enumerated(EnumType.STRING)
    val role : RoleType,

    @OneToMany(cascade = [CascadeType.ALL], orphanRemoval = true)
    val boards: MutableList<Board>? = mutableListOf(),

    @Id
    @GeneratedValue(strategy = IDENTITY)
    private val id: Long? = null
) {


}

enum class RoleType {
    ROLE_USER,
    ROLE_ADMIN
}
